src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java

```diff
@@ -0,0 +1,329 @@
+ *    * &#64;param array desc
+ *    * &#64;see Some javadoc      // OK
+ *    * &#64;param e desc          // violation, wrong order for "e" tag
+ *    * &#64;return Some number.
+ *    *&#47;
+ *   public static &lt;T, E, S&gt; int sampleMethod(T[] array, String str, E e) {
+ *     return 2021;
+ *   }
+ *
+ *   &#47;**
+ *    * No violation for missing "output" parameter.
+ *    *
+ *    * &#64;param input desc      // OK
+ *    *&#47;
+ *   public void sampleMethod2(String input, String output) {
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * This check will not report any violations for extra parameter in javadoc.
+ * </p>
+ * <pre>
+ * class example3 {
+ *   &#47;**
+ *    * No violation for extra "extra" parameter.
+ *    *
+ *    * &#64;version Some javadoc.
+ *    * &#64;param &lt;T&gt; description
+ *    * &#64;param E desc          // violation, wrong order for "E" tag
+ *    * &#64;param &lt;S&gt; desc
+ *    * &#64;param array desc
+ *    * &#64;see Some javadoc      // OK
+ *    * &#64;param e desc          // violation, wrong order for "e" tag
+ *    * &#64;param str desc        // violation, wrong order for "str" tag
+ *    * &#64;param extra desc      // OK, Extra parameter
+ *    * &#64;return Some number
+ *    *&#47;
+ *   public static &lt;T, E, S&gt; int sampleMethod(T[] array, String str, E e ) {
+ *     return 2021;
+ *   }
+ *
+ *   &#47;**
+ *    * No violation for extra "t" parameter.
+ *    *
+ *    * &#64;param t desc          // OK, Extra parameter
+ *    *&#47;
+ *   public void sampleMethod2() {}
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+  * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.order}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.42
+ */
+@StatelessCheck
+public class JavadocParamOrderCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_TAG_ORDER = "javadoc.tag.order";
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode node) {
+
+        final DetailAST ast = getBlockCommentAst();
+        if (ast.getType() == TokenTypes.METHOD_DEF
+                 || ast.getType() == TokenTypes.CTOR_DEF) {
+            final List<DetailAST> methodParams = getMethodParameters(ast);
+            final List<DetailAST> methodTypeParams = CheckUtil.getTypeParameters(ast);
+
+            checkParamsOrder(node, methodParams, methodTypeParams);
```

Comment:
@aryaniiit002 variable `ast` is always of type `BLOCK_COMMENT_BEGIN`. Your code isn't working as it should be `METHOD_DEF` or `CTOR_DEF` according to your check.

Reply:
i think we need a loop before `if` here.

Reply:
what exactly does `getBlockCommentAst()` do?
i thought it gives ast like we get in visitToken()

Reply:
how do i get correct `ast` to get method params?

Reply:
i want to know how can i access ast in visitJavadocToken() bcoz check requires method parameters and it require ast..
i tried using getBlockCommentAst() with getParent() but it's showing getBlockCommentAst is null

Reply:
i want to know how can i access ast in visitJavadocToken() bcoz check requires method parameters and it require ast..
i tried using getBlockCommentAst() with getParent() but it's showing `getBlockCommentAst().getParent()` is null

Reply:
I had done a PR to your branch. Please have a look.

Reply:
resolved